internal/credentials/kubernetes/github/app_test.go

```diff
@@ -40,7 +40,25 @@ func TestAppCredentialHelper(t *testing.T) {
 		assertions func(*testing.T, *credentials.Credentials, *cache.Cache, error)
 	}{
 		{
-			name:     "cred type is not git",
+			name:     "cred type is not supported",
+			credType: credentials.TypeHelm,
+			helper:   &appCredentialHelper{},
+			assertions: func(t *testing.T, creds *credentials.Credentials, _ *cache.Cache, err error) {
+				require.NoError(t, err)
+				require.Nil(t, creds)
+			},
+		},
+		{
+			name:     "cred type is git",
+			credType: credentials.TypeGit,
+			helper:   &appCredentialHelper{},
+			assertions: func(t *testing.T, creds *credentials.Credentials, _ *cache.Cache, err error) {
+				require.NoError(t, err)
+				require.Nil(t, creds)
+			},
+		},
+		{
+			name:     "cred type is image",
```

Comment:
These two new cases are still not needed. I know this may seem counter-intuitive, but remember this is white-box testing...

The previous case (cred type not supported) tests for an early return if the cred type is not supported. Subsequent cases only require the cred type _to be a supported one_ so that we avoid that early return and move on to testing other outcomes. Again, with white-box testing, we have the luxury of knowing the implementation and we know that once we're past that potential early return, the cred type has no further bearing on the results.